src/OpenApi/src/Services/OpenApiComponentService.cs

```diff
@@ -12,27 +22,50 @@ namespace Microsoft.AspNetCore.OpenApi;
 /// an OpenAPI document. In particular, this is the API that is used to
 /// interact with the JSON schemas that are managed by a given OpenAPI document.
 /// </summary>
-internal sealed class OpenApiComponentService
+internal sealed class OpenApiComponentService(IOptions<JsonOptions> jsonOptions)
 {
-    private readonly ConcurrentDictionary<Type, OpenApiSchema> _schemas = new()
+    private readonly ConcurrentDictionary<(Type, ParameterInfo?), JsonObject> _schemas = new()
```

Comment:
AFAIK, `ValueTuples` have safe semantics for use as dictionary keys (equality checks, hash codes, etc.). I can take a look at the perf comparison but my bet is that it might be small given `ValueTuples` are structs under the hood?

Reply:
Good enough for me.